### PR TITLE
DataDesign.Common and PerformanceProvider are package references rather than referenced from disk

### PR DESCRIPTION
**Pull request by basoundr**
Tagging @jmarolf @srivatsn  @davkean for review

**Customer scenario**

Project system signed builds are running into issues because of failure to find “Microsoft.VisualStudio.DataDesign.Common” and "Microsoft.VisualStudio.Diagnostics.PerformanceProvider"in the machine. Hence we get the references from Package rather than from disk.

**Risk**

Low- This is just infra change and no code change.

**Performance impact**

None.

### CODE REVIEW

**Comment by jmarolf** [comment]
Shouldn't block merging this in, but we should make sure that all of our Visual Studio dependencies are package references.  Most if not all of the others should already have packages

**Comment by basoundr** [comment]
Adding @jasonmalinowski

**Comment by basoundr** [comment]
@dotnet-bot retest Windows Release please

> 13:51:25   Restoring packages for ProjectSystem (this may take some time)
> 13:51:46 C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets(97,5): error : An error occurred while retrieving package metadata for 'RoslynDependencies.OptimizationData.2.0.0-rc-61101-17' from source 'C:\Users\dotnet-bot\.nuget\packages\'. [D:\j\workspace\windows_relea---22c57ed6\src\ProjectSystem.sln]
> 13:51:46 C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets(97,5): error :   '.', hexadecimal value 0x00, is an invalid character. Line 1, position 1. [D:\j\workspace\windows_relea---22c57ed6\src\ProjectSystem.sln]

File: build/Targets/References.targets [hunk 1]
```diff
@@ -116,14 +116,10 @@
...
-          <HintPath Condition="Exists('C:\Program Files (x86)\Microsoft Visual Studio\Preview\Professional\Common7\IDE\PrivateAssemblies\Microsoft.VisualStudio.Diagnostics.PerformanceProvider.dll')">C:\Program Files (x86)\Microsoft Visual Studio\Preview\Professional\Common7\IDE\PrivateAssemblies\Microsoft.VisualStudio.Diagnostics.PerformanceProvider.dll</HintPath>
-          <HintPath Condition="Exists('C:\Program Files (x86)\Microsoft Visual Studio\Preview\Community\Common7\IDE\PrivateAssemblies\Microsoft.VisualStudio.Diagnostics.PerformanceProvider.dll')">C:\Program Files (x86)\Microsoft Visual Studio\Preview\Community\Common7\IDE\PrivateAssemblies\Microsoft.VisualStudio.Diagnostics.PerformanceProvider.dll</HintPath>
-        </Reference>
 
         <PackageReference Include="Microsoft.VSSDK.BuildTools" Version="15.0.26124-rc3" />
+        <PackageReference Include="Microsoft.VisualStudio.DataDesign.Common" Version="15.0.26524-alpha" />
```
Comment (jmarolf) [inline build/Targets/References.targets]: Why was this moved here instead of staying in `IncludeVisualStudioDesignerReferences`?

**Comment by basoundr** [comment]
Closing this PR since #2327 made the same changes
